packages/beacon-node/test/unit/network/peers/client.test.ts

```diff
@@ -32,7 +32,7 @@ describe("clientFromAgentVersion", () => {
 
   for (const {name, agentVersion, client} of testCases) {
     it(name, () => {
-      expect(clientFromAgentVersion(agentVersion)).toBe(client);
+      expect(getKnownClientFromAgentVersion(agentVersion)).toBe(client);
```

Comment:
Worth adding a test validating that `agentVersion` would be used if `agentVersion` is null?